playback/tape_cassettes/s3/s3_tape_cassette.py

```diff
@@ -259,6 +259,12 @@ def content_filter_func(recording_str):
                 if isinstance(v, str):
                     if not fnmatch(recorded_value, v):
                         return False
+                elif isinstance(v, list):
+                    for value in v:
+                        if fnmatch(recorded_value, value):
+                            break
```

Comment:
Doesn't it mean that if nothing matches you return True?
I assume this is not what you mean and you should replace lines 264-267 with:
return any(fnmatch(recorded_value, value) for value in v) no?
Also please update your test so it will fail on this case before you do this fix

Reply:
for else means that if nothing is found (didn't reach the break) it goes to the else:
https://book.pythontips.com/en/latest/for_-_else.html#:~:text=This%20is%20the%20basic%20structure%20of%20a%20for/else%20loop%3A